### ISSUE TITLE
[feature] Add 1:1 "to-scale" printing output or image creation, for peg-board layout generation

### ISSUE THREAD

**Issue by tim292stro** [formatc1702/WireViz#61]
In higher volume production environments, a harness that will be produced several times is generally laid out on a peg board.  This allows wires to be cut in-place at the correct length, and laid from point-to-point before bundling and insertion into the connector.
![automotive_wireharness](https://user-images.githubusercontent.com/45922790/86518129-a57ed500-bde3-11ea-876b-5535ab6ccf9e.jpg)

One normally prints out a harness on a large format printer, then lays the diagram on an appropriately sized material.  This can be done on either a piece of wood or metal, normally pegs or nails can be inserted at fan-out points, allowing the wires to be separated.  This can also be done on a magnetic material and using hanger magnets (magnets with hooks attached).
![A620-Peg-Board 2](https://user-images.githubusercontent.com/45922790/86518181-335ac000-bde4-11ea-9123-0f4b8b0dbbdc.jpg)

For quality control purposes, these peg boards often have the mating connector for end terminations so that once completed the entire harness can be fault-checked.  Continuity checks

For a short-run/bespoke or doing several different harnesses per day in the same station, having a 1:1 scale model projected on a white metallic surface and using repositionable magnet hooks would allow the same effect without a large format printer.  Having a known scale in X/Y axis dimensions allows external math to be done.  For example having a 4k (3840x2160) projector throwing a 170.67"x96" 1:1 scale image on a white-painted steel table top we'd know each pixel was 0.044"x0.044".  That's a 14.25'x8' table.  This would require the image generation to consider the constraints of the output dimensions.  Note that on this example:
![peg-board-testing-1](https://user-images.githubusercontent.com/45922790/86518148-dc54eb00-bde3-11ea-9701-9aa55b41b333.png)
some legs of the harness are wrapped around to fit in the space allowed.

**Comment by aakatz3** [comment]
This is a very interesting problem to solve. Are these usually created by hand, and how are the paths laid out? It might be possible, using the bundles and a new output stage for the harness, to generate paths with specific lengths, and "length tuning" to make sure the lengths take up minimal physical space.

**Comment by tim292stro** [comment]
Normally these are done in CAD with a (rather expensive) 3rd party tool (like this from Mentor Graphics: "https://www.mentor.com/products/electrical-design-software/capital-harness-xc" - I believe it's ~$2K-USD/seat IIRC).  The tool takes into account the destination page size then wraps them if needed to splay them out without overlapping.  Think something like a big hydra-level game of "Snake".

Until recently, I had been using the side of a steel container and some hook magnets (like: https://www.amazon.com/BAVITE-Magnetic-Neodymium-Kitchen-Workplace/dp/B081WVM17B) to do layouts on a wall.  Just used a notepad and a tape measure, but the setup and QC checking time was a killer.

The critical detail is getting the lengths between the junctions and the ends of the bundles correct - then having the correct lengths for each connector wire (all of this is already captured in the text files from what I can read).

**Comment by formatc1702** [comment]
While it does sound very interesting, I doubt we'll be seeing this feature in WireViz anytime soon... since GraphViz loves moving nodes around, and it's hard enough to get it to display nicely at all... it's optimized to display logical elements and connections among them, not a physical layout.
But, one can dream :)

**Comment by tim292stro** [comment]
Yes, I realized that GraphViz is limiting or making it necessary to work around (more than just this item) - this is more of a wish-list I guess should one be considering porting another graphic back-end API.

**Comment by aakatz3** [comment]
As a separate note, if you have Microsoft Visio, and GraphVisio, you may be able to take the wireviz output, and then possibly lay it out to-scale in that? Visio is expensive, and GraphVisio is somewhat buggy, but it is an option!
